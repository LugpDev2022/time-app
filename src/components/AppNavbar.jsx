import { Container, Nav, Navbar } from 'react-bootstrap';
import { NavbarBrand, NavbarLink } from './';

export const AppNavbar = () => (
  <Navbar expand='md' bg='dark' variant='dark'>
    <Container>
      <NavbarBrand />
      <Navbar.Toggle aria-controls='navbar-nav' />
      <Navbar.Collapse id='navbar-nav'>
        <Nav>
          <NavbarLink to='/'>Clock</NavbarLink>
          <NavbarLink to='/world-clock'>World Clock</NavbarLink>
          <NavbarLink to='/timer'>Timer</NavbarLink>
        </Nav>
      </Navbar.Collapse>
    </Container>
  </Navbar>
);
