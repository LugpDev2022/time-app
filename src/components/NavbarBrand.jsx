import Navbar from 'react-bootstrap/Navbar';
import { AiFillClockCircle } from 'react-icons/ai';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';

export const NavbarBrand = () => {
  const { isRunning } = useSelector(state => state.timer);

  if (isRunning) {
    return (
      <Navbar.Brand>
        <span className='text-muted fs-4 fw-bold d-flex align-items-center gap-1'>
          <AiFillClockCircle /> Time App
        </span>
      </Navbar.Brand>
    );
  }

  return (
    <Navbar.Brand>
      <Link
        to='/'
        className='text-decoration-none text-primary fs-4 fw-bold d-flex align-items-center gap-1'
      >
        <AiFillClockCircle /> Time App
      </Link>
    </Navbar.Brand>
  );
};
