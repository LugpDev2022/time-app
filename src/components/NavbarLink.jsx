import { NavLink, useLocation } from 'react-router-dom';

export const NavbarLink = ({ to, children }) => {
  const { pathname } = useLocation();

  return (
    <NavLink
      className={`text-primary fs-5 nav-link ${
        pathname === to ? 'text-decoration-underline' : 'text-decoration-none'
      }`}
      to={to}
    >
      {children}
    </NavLink>
  );
};
