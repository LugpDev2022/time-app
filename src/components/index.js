export * from './AppNavbar';
export * from './NavbarBrand';
export * from './NavbarLink';
export * from './styled';
