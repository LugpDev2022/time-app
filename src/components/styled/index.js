export * from './BigNumber';
export * from './Box';
export * from './Title';
