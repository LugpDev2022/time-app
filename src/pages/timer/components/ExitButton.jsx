import { ImExit } from 'react-icons/im';
import { Link } from 'react-router-dom';

export const ExitButton = () => {
  return (
    <Link to='/timer'>
      <ImExit className='text-danger fs-4 position-absolute top-0 left-0 mt-4 ms-1 translate-middle cursor-pointer hover' />
    </Link>
  );
};
