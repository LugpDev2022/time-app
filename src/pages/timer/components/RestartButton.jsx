import { VscDebugRestart } from 'react-icons/vsc';
import { useDispatch, useSelector } from 'react-redux';
import { resetTimer } from '../../../store/slices/timer/timerSlice';

export const RestartButton = () => {
  const { isFinished } = useSelector(state => state.timer);
  const dispatch = useDispatch();

  const handleClick = () => {
    dispatch(resetTimer());
  };


  if (isFinished) {
    return (
      <div className='position-absolute top-50 start-0 translate-middle ms-4'>
        <VscDebugRestart className='text-danger fs-2' onClick={handleClick} />
      </div>
    );
  }
};
