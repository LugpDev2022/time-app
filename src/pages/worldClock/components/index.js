export * from './AddCountryForm';
export * from './CountryClockItem';
export * from './CountryClockList';
