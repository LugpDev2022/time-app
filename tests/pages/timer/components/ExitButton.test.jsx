import React from "react";
import { render, screen } from "@testing-library/react";
import { ExitButton } from "../../../../src/pages/timer/components";
import { MemoryRouter } from "react-router-dom";

describe("tests on <ExitButton />", () => {
  test("should match snapshot", () => {
    const { container } = render(
      <MemoryRouter>
        <ExitButton />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });

  test("should redirect to '/timer'", () => {
    render(
      <MemoryRouter>
        <ExitButton />
      </MemoryRouter>
    );

    const link = screen.getByRole("link");
    expect(link.href).toContain("/timer");
  });
});
