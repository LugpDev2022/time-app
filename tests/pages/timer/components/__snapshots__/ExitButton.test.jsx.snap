// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <ExitButton /> should match snapshot 1`] = `
<div>
  <a
    href="/timer"
  >
    <svg
      class="text-danger fs-4 position-absolute top-0 left-0 mt-4 ms-1 translate-middle cursor-pointer hover"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      version="1.1"
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 10v-2h-5v-2h5v-2l3 3zM11 9v4h-5v3l-6-3v-13h11v5h-1v-4h-8l4 2v9h4v-3z"
      />
    </svg>
  </a>
</div>
`;
